driver/src/orderbook/mod.rs

```diff
@@ -62,7 +62,9 @@ impl OrderbookReaderKind {
                 auction_data_page_size,
                 orderbook_filter,
             )),
-            OrderbookReaderKind::EventBased => Box::new(EventBasedOrderbook::new(contract, web3)),
+            OrderbookReaderKind::EventBased => {
+                Box::new(EventBasedOrderbook::new(contract, web3).unwrap())
```

Comment:
I suppose this was a mistake from the original PR.

Reply:
No mistake in the original PR. The function signature of `new` has changed so that it returns a Result.